x-pack/plugins/security_solution/cypress/tasks/api_calls/rules.ts

```diff
@@ -51,98 +55,107 @@ export const createCustomRule = (rule: CustomRule, ruleId = 'rule_testing', inte
     failOnStatusCode: false,
   });
 
-export const createEventCorrelationRule = (rule: CustomRule, ruleId = 'rule_testing') =>
-  cy.request({
-    method: 'POST',
-    url: 'api/detection_engine/rules',
-    body: {
-      rule_id: ruleId,
-      risk_score: parseInt(rule.riskScore, 10),
-      description: rule.description,
-      interval: `${rule.runsEvery.interval}${rule.runsEvery.type}`,
-      from: `now-${rule.lookBack.interval}${rule.lookBack.type}`,
-      name: rule.name,
-      severity: rule.severity.toLocaleLowerCase(),
-      type: 'eql',
-      index: rule.index,
-      query: rule.customQuery,
-      language: 'eql',
-      enabled: true,
-    },
-    headers: { 'kbn-xsrf': 'cypress-creds' },
-  });
+export const createEventCorrelationRule = (rule: CustomRule, ruleId = 'rule_testing') => {
+  if (rule.dataSource.type === 'indexPatterns') {
```

Comment:
Instead of having this condition, let's support both index patterns and data views here as in `createCustomRule`.

Reply:
I would prefer to do this on a different PR, since it could be a bit complex and at least we will have a test for data views already in main.

Reply:
That sounds good to me, thanks @MadameSheema 👍